src/js/legend.js

```diff
@@ -25,19 +25,61 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             // dataSet accepts:
             // 1. an array of objects. Must contain "id", "value" and "label" variables.
             // Example: [{id: string, value: number, label: string} ... ]
-            dataSet: []
+            dataSet: [],
+            total: {
+                // value: number,
+                // percentage: number
+            }
         },
+        modelRelay: [{
```

Comment:
Per @amb26's suggestion from the discussion with @cindyli , I've done the following in a3c7d2b:
 - created a separate totalRelaying grade containing the common modelRelay block used between the dataEntryPanel, pie and legend components
 - used this grade in the gradenames of those three components

Works great!